spec.go

```diff
@@ -0,0 +1,13 @@
+package crdstorage
+
+type deletionConfig struct {
+	Spec deletionConfigSpec `json:"spec"`
+}
+
+type deletionConfigSpec struct {
+	Storage deletionConfigSpecStorage `json:"storage" yaml:"storage"`
+}
+
+type deletionConfigSpecStorage struct {
+	Data map[string]*string `json:"data" yaml:"data"`
+}
```

Comment:
The inline struct magic did not work out somehow. So this is necessary for the deletion magic.

Reply:
Can we name the file types.go? My understanding is a  spec.go is for the interface. 

Also deletionConfig is misleading. I'd rather call it storageConfigJsonPatch. If we have separate type let's maybe use it for both Put and Delete with a comment why we need to use `*string`. WDYT?

Reply:
Done.